integrations/google_genai/src/haystack_integrations/components/generators/google_genai/chat/chat_generator.py

```diff
@@ -397,82 +397,51 @@ def _process_streaming_chunk(
             candidate = chunk.candidates[0]
             finish_reason = getattr(candidate, "finish_reason", None)
 
-            if candidate.content and candidate.content.parts:
-                tc_index = -1
-                for part in candidate.content.parts:
-                    if part.text:
-                        all_text_parts.append(part.text)
-                        content += part.text
+        usage_metadata = chunk.usage_metadata
 
-                    elif part.function_call:
-                        tc_index += 1
-                        tool_call = ToolCall(
-                            tool_name=part.function_call.name or "",
-                            arguments=part.function_call.args if part.function_call.args else {},
+        usage = {
+            "prompt_tokens": getattr(usage_metadata, "prompt_token_count", 0) if usage_metadata else 0,
+            "completion_tokens": getattr(usage_metadata, "candidates_token_count", 0) if usage_metadata else 0,
+            "total_tokens": getattr(usage_metadata, "total_token_count", 0) if usage_metadata else 0,
+        }
+
+        if candidate.content and candidate.content.parts:
+            tc_index = -1
+            for part in candidate.content.parts:
+                if part.text:
+                    content += part.text
+
+                elif part.function_call:
+                    tc_index += 1
+                    tool_calls.append(
+                        ToolCallDelta(
+                            # Google GenAI does not provide index, but it is required for tool calls
+                            index=tc_index,
                             id=part.function_call.id,
+                            tool_name=part.function_call.name or "",
+                            arguments=json.dumps(part.function_call.args) if part.function_call.args else None,
                         )
-                        all_tool_calls.append(tool_call)
-                        tool_calls.append(
-                            ToolCallDelta(
-                                # Google GenAI does not provide index, but it is required for tool calls
-                                index=tc_index,
-                                id=tool_call.id,
-                                tool_name=tool_call.tool_name,
-                                arguments=json.dumps(tool_call.arguments) if tool_call.arguments else None,
-                            )
-                        )
+                    )
+
+        # start is only used by print_streaming_chunk. We try to make a reasonable assumption here but it should not be
+        # a problem if we change it in the future.
+        start = index == 0 or len(tool_calls) > 0
 
-        # Prioritize tool calls over content when both are present
         return StreamingChunk(
-            content="" if tool_calls else content,
+            content="" if tool_calls else content,  # prioritize tool calls over content when both are present
```

Comment:
Is it possible for both to be present at the same time?

Reply:
generally, this does not happen for Google